extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/conf/ConfigurationService.java

```diff
@@ -321,6 +322,26 @@ public ExprNode getUserSearchFilter() throws GuacamoleException {
         );
     }
 
+    /**
+     * Returns the search filter that should be used when querying the
+     * LDAP server for Guacamole groups.  If no filter is specified,
+     * a default of "(objectClass=group)" is returned.
+     *
+     * @return
+     *     The search filter that should be used when querying the
+     *     LDAP server for groups that are valid in Guacamole, or
+     *     "(objectClass=group)" if not specified.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public ExprNode getGroupSearchFilter() throws GuacamoleException {
+        return environment.getProperty(
+            LDAPGuacamoleProperties.LDAP_GROUP_SEARCH_FILTER,
+            new EqualityNode("objectClass","group")
```

Comment:
I think @mike-jumper's vote here was to make the default the `PresenceNode("objectClass")`, similar to how users are done.  I'm good with either one, but it does seem more consistent at present to stick with that.

Reply:
I think he observed the change to pre-existing ldap-user-search-filter function, that it's out of the scope of this GUACAMOLE-996

Reply:
Yes, modifying the user one is certainly out-of-scope - I'm not just not sure if we should be changing the default for user searches at this point vs. providing a way for people to change it if they want to.  In the end the question becomes, "How many confused admins are we going to have out there when they upgrade from 1.2.0 to 1.3.0 and all of the sudden their group searches don't work as expected?"  Changing defaults should be done with care.

Reply:
@mike-jumper: Thoughts on this?  The above change will essentially change the default group filter from `(objectClass=*)` to `(objectClass=group)`.  While I suspect that's probably okay and what we want in the long-run, I worry about catching unsuspecting admins off-guard...

Reply:
I think we should lean toward Principle of Least Surprise and maintain the existing `(objectClass=*)` behavior by default. The default behavior is intended to rely on LDAP directory structure to determine which objects should be returned and classified as users or groups, and I think it makes sense to continue to use that as the default behavior.

Filters like `ldap-user-search-filter` and soon-to-be-implemented `ldap-group-search-filter` exist to allow for more complex and intermingled cases, but default filters would involve too many assumptions and could break existing deployments.